src/AutoStaker.ts

```diff
@@ -260,34 +250,54 @@ export default class AutoStaker {
               contract_addr: assetToken.token.contractAddress as string
             }
           },
-          amount: new Int(assetProvideAmount).toString()
+          amount: new Int(assetProvideAmount).toString(),
         },
         {
           info: {
             native_token: {
               denom: 'uusd'
             }
           },
-          amount: new Int(uusdProvideAmount).toString()
+          amount: new Int(uusdProvideAmount).toString(),
         }
       ])
     ]);
+    console.log(`  >> Provided ${toDecimal(assetProvideAmount)}${TARGET_ASSET} & ${toDecimal(uusdProvideAmount)}UST`);
 
-    const lpTokenBlanace = await assetToken.lpToken.getBalance();
-
-    console.log('Stake LP token');
+    console.log('Staking LP token');
+    const lpTokenBalance = await assetToken.lpToken.getBalance();
     await this.execute([
       this.mirror.staking.bond(
-        assetTokenAddr,
-        lpTokenBlanace.balance,
+        this.assetTokenAddr,
+        lpTokenBalance.balance,
         assetToken.lpToken
       )
     ]);
+    console.log(`  >> Staked ${toDecimal({ d: [lpTokenBalance.balance] })} LP to ${TARGET_ASSET}-UST LP`);
 
     console.log('Done');
   }
+  async rewardsToClaim() {
+    const poolInfo = await this.mirror.staking.getPoolInfo(this.assetTokenAddr);
+    const rewardInfoResponse = await this.mirror.staking.getRewardInfo(
+      this.wallet.key.accAddress,
+      this.assetTokenAddr
+    );
+    const hasRewards = poolInfo.reward_index !== rewardInfoResponse.reward_infos[0].index;
```

Comment:
Also need to check pending_reward

Reply:
With the 15 second delay between contract executions there should never be pending rewards. Though I'm unfamiliar with how I might access info on pending_rewards.

Reply:
Understood